hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/utilops/ProviderRun.java

```diff
@@ -193,12 +204,23 @@ protected boolean submitOp(HapiSpec spec) {
             } else {
                 log.warn("Now {} ops pending; backing off for {}s!", numPending, BACKOFF_SLEEP_SECS);
                 try {
-                    Thread.sleep(BACKOFF_SLEEP_SECS * 1_000L);
+                    sleep(BACKOFF_SLEEP_SECS * 1_000L);
                 } catch (InterruptedException ignore) {
                 }
             }
         }
 
+        if (waitForPendingOps) {
+            if (!loggingOff) {
+                log.info("Waiting for {} pending ops to finish...", spec.numPendingOps());
+            }
+            long beginWait = Instant.now().toEpochMilli();
+            while (spec.numPendingOps() > 0
+                    || (Instant.now().toEpochMilli() - beginWait) < WAIT_PENDING_OPS_TIMEOUT.toMillis()) {
+                onSpinWait();
+            }
+        }
```

Comment:
Is there nothing else we can trigger off of, instead of just waiting? Sometimes a spin loop is necessary, yes, but we try to avoid it unless absolutely needed